### PR TITLE
Improve publication page UI

### PR DESCRIPTION
**Pull request by pmyip3**
# Description 
Improve publication page UI

# Type of change 
- [ ] Bug fix
- [ ] New feature
- [x] Refactoring
- [ ] Documentation 
- [x] Improvement 

# Problem
Current UI does not look good

# Solution description
1. Add border around current publications page editor
2. Refactor "down" arrow in publication details
3. Refactor styling of button layouts in publication page editor
4. Add checkbox for delete batch publications functionality (actual implementation will be in a separate ticket)
5. Button to select all/none publications
6. Replace the page add button with 'Add' text

- also change pagination style 

Refer to: https://trello.com/c/0wO9gkQU/125-improve-publication-page-ui

# Tests 
All unit tests pass? N/A

Builds successfully? Yes

Attach screenshot of the expected result (e.g. UI of the website/ postman api call result/ database changes)
image
image
image

Provide steps in detail for reviewers to reproduce the expected result of the code changes.
The functionality of the publication page should work as before. 
Try to add/import/edit/delete publications to see if they work as expected. 

# Relevant document/ link (if any) 
https://trello.com/c/0wO9gkQU/125-improve-publication-page-ui

# Risk
Low, only change the UI component

### CODE REVIEW

**Comment by vgoh1** [comment]
Could we try with the title of the publication as black so it stands out more, and the fields (authors etc) in the grey? Similarly with the Group By/Sort By buttons, maybe the unselected text could be grey on a white background?

File: client/src/components/publications/publicationsLayout/PublicationsEditor.js [hunk 1]
```diff
@@ -0,0 +1,142 @@
...
+        </EditorButton>
+      </StyledButtonGroup>
+    </>
+  );
+};
+// props validation
+PublicationsEditor.propTypes = {
+  options: PropTypes.object.isRequired,
+  setOptions: PropTypes.func.isRequired,
+  publication: PropTypes.object.isRequired,
```
Comment (jfan0010) [inline client/src/components/publications/publicationsLayout/PublicationsEditor.js:136]: I think publication is an array now? might need to change this validation.
image

**Review by jfan0010 — APPROVED** [review]
The UI looks good to me.
Just curious, is there a reason we're using buttons for `Group By` and `Sort By` options, instead of others (like dropdown list/radio buttons etc.)?

**Comment by pmyip3** [comment]
> Could we try with the title of the publication as black so it stands out more, and the fields (authors etc) in the grey? Similarly with the Group By/Sort By buttons, maybe the unselected text could be grey on a white background?

Sure, will make different color combinations and put them on the PR.

**Comment by pmyip3** [comment]
> The UI looks good to me.
> 
> Just curious, is there a reason we're using buttons for `Group By` and `Sort By` options, instead of others (like dropdown list/radio buttons etc.)?

No specific reason 🙈
But I think the dropdown require extra click from user so I changed it. Radio buttons would work too, but I reckon it's easier for users to click on the options directly on button instead?!
It would be an issue if we have a few more group by/ sorting options in the future tho.

File: client/src/components/publications/publicationsLayout/PublicationsEditor.js (hunk 1 above)
Comment (pmyip3) [reply client/src/components/publications/publicationsLayout/PublicationsEditor.js:136]: Thank you for that!

**Comment by vgoh1** [comment]
Thanks for making the changes! Just one thing I noticed while testing:

So if I select all publications when it's grouped by all publications, and then group by category, shouldn't the box still stay ticked?
image
This is the same case for the other categories as well, where the box isn't ticked.

Also in the button I prefer Group By: `All Publications` rather `All Publication`

**Comment by pmyip3** [comment]
> Thanks for making the changes! Just one thing I noticed while testing:
> 
> So if I select all publications when it's grouped by all publications, and then group by category, shouldn't the box still stay ticked?
> image
> This is the same case for the other categories as well, where the box isn't ticked.
> 
> Also in the button I prefer Group By: `All Publications` rather `All Publication`
Should be all good now. 
And I just merge from main, so the Group by options should be 'None' and 'Category'.

**Review by vgoh1 — APPROVED** [review]

**Comment by cyr17** [comment]
image

Do you reckon we should have a fixed border? because say if I have a single publication the border spans just half the page length but having this border fixed with a small padding at the edges would be better to look at

File: client/src/components/publications/publicationsLayout/PublicationsEditor.js [hunk 2]
```diff
@@ -0,0 +1,146 @@
...
+            Add
+          </Dropdown.Toggle>
+          <Dropdown.Menu>
+            <Dropdown.Item onClick={() => setShowCreateForm(true)}>Insert Publication</Dropdown.Item>
+            <Dropdown.Item onClick={() => setShowImportForm(true)}>Import Publications</Dropdown.Item>
+          </Dropdown.Menu>
+        </Dropdown>
+      </StyledButtonGroup>
+      <StyledButtonGroup>
+        Group By
```
Comment (cyr17) [inline client/src/components/publications/publicationsLayout/PublicationsEditor.js:81]: Maybe have some padding in between these filters, rn it's cluttered , so maybe space it evenly and have the update layout at the edge

Comment (pmyip3) [reply client/src/components/publications/publicationsLayout/PublicationsEditor.js:81]: > image
> 
> Idk if this is a bug from before! But the author field won't let you enter multiple characters at once.

Thanks for catching this, not sure when this bug exists but I will fix this in this PR.

**Comment by pmyip3** [comment]
> image
> 
> Idk if this is a bug from before! But the author field won't let you enter multiple characters at once.

Thanks for catching this, not sure when this bug exists but I will fix this in this PR.

**Review by aksh0001 — COMMENTED** [review]
Could you rollback the changes to the package lock for the API, seeing as we’re not introducing dependency changes?

**Comment by kdav108** [comment]
Great work on refactoring the entire UI Mable!

I have a few suggestions (probably not worth implementing all of them given our Short deadline), but here they are in order of my preference:
1) Selecting all and clicking on the delete button isn't deleting anything. Not sure if this is just on my end.

2) When the user has no publications, the overall UI looks a bit too cluttered. Would be nice to move the message down (perhaps half way down?) Considering all our users will see this at the beginning, it would be nice to make it look better.
Screen Shot 2021-09-18 at 5 23 11 pm

3) When you select `group by` to `Category` the UI of the entire row changes (to accomodate the `Category Title` option perhaps?) If it's possible fixing it in place would be nice.

Screen Shot 2021-09-18 at 5 44 33 pm
**V/S**
Screen Shot 2021-09-18 at 5 23 39 pm

4) How about renaming 'Insert Publication' to 'Add Manually'? because importing is also 'inserting'

Screen Shot 2021-09-18 at 5 24 00 pm


Feel free to entertain the ones that we feel more important

**Comment by pmyip3** [comment]
> Great work on refactoring the entire UI Mable!
> 
> I have a few suggestions (probably not worth implementing all of them given our Short deadline), but here they are in order of my preference:
> 
> 1. Selecting all and clicking on the delete button isn't deleting anything. Not sure if this is just on my end.
> 2. When the user has no publications, the overall UI looks a bit too cluttered. Would be nice to move the message down (perhaps half way down?) Considering all our users will see this at the beginning, it would be nice to make it look better.
> 
> Screen Shot 2021-09-18 at 5 23 11 pm
> 
> 1. When you select `group by` to `Category` the UI of the entire row changes (to accomodate the `Category Title` option perhaps?) If it's possible fixing it in place would be nice.
> 
> Screen Shot 2021-09-18 at 5 44 33 pm
> 
> **V/S**
> Screen Shot 2021-09-18 at 5 23 39 pm
> 
> 1. How about renaming 'Insert Publication' to 'Add Manually'? because importing is also 'inserting'
> 
> Screen Shot 2021-09-18 at 5 24 00 pm
> 
> Feel free to entertain the ones that we feel more important

1. It would be in another ticket as this PR is pretty big already, hopefully, can get that done tmr. 
 Have fixed the rest of the suggestions. Free feel to test it again :)

**Comment by pmyip3** [comment]
> Could you rollback the changes to the package lock for the API, seeing as we’re not introducing dependency changes?

I have reverted the file but it still got around 300 changes. Is it normal?

**Review by kdav108 — APPROVED** [review]
LGTM!

**Comment by kdav108** [comment]
> > Could you rollback the changes to the package lock for the API, seeing as we’re not introducing dependency changes?
> 
> I have reverted the file but it still got around 300 changes. Is it normal?

Don't think it should show that, did u revert the file and push the commit?

**Comment by pmyip3** [comment]
> > > Could you rollback the changes to the package lock for the API, seeing as we’re not introducing dependency changes?
> > 
> > 
> > I have reverted the file but it still got around 300 changes. Is it normal?
> 
> Don't think it should show that, did u revert the file and push the commit?

**Pull request merged**